connectivity/drivers/wifi/CMakeLists.txt

```diff
@@ -1,4 +1,10 @@
 # Copyright (c) 2020 ARM Limited. All rights reserved.
 # SPDX-License-Identifier: Apache-2.0
 
+add_library(mbed-os-wifi-esp8266 OBJECT)
```

Comment:
I think you should create this target library at the subdirectory `./esp8266-driver/` and not here as we may add other WiFi drivers.

Reply:
I'll fix